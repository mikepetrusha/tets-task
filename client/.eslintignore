node_modules
dist
coverage
./babel.config.json
./jest.config.js