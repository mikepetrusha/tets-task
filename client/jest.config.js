/** @type {import('jest').Config} */
const config = {
  collectCoverage: true,
  collectCoverageFrom: ['src/**/*.{js,ts,jsx,tsx}'],
  coverageDirectory: 'coverage',
  testEnvironment: 'jsdom',
  setupFilesAfterEnv: ['<rootDir>/setup-tests.js'],
  transform: {
    '^.+\\.(js|jsx|ts|tsx)$': 'babel-jest'
  },
  moduleFileExtensions: ['js', 'ts', 'jsx', 'tsx'],
  moduleNameMapper: {
    '^.+\\.svg$': 'jest-svg-transformer',
    '^.+\\.css$': 'identity-obj-proxy',
    '^@/(.*)$': '<rootDir>/src/$1'
  }
};

export default config;
