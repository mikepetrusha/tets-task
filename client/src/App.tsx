import { RouterProvider } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyle } from '@/config';
import theme from '@/config/theme/theme';
import { router } from '@/router';

export const App = () => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      <RouterProvider router={router} />
    </ThemeProvider>
  );
};
