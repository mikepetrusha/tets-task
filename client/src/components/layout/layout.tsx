import { Suspense } from 'react';
import { Outlet } from 'react-router-dom';

import { DialogsContainer } from '../dialogs-container';
import { Header } from '../header';
import { Spinner } from '../ui';

export const Layout = () => {
  return (
    <>
      <Header />
      <main>
        <Suspense fallback={<Spinner />}>
          <Outlet />
        </Suspense>
      </main>
      <DialogsContainer />
    </>
  );
};
