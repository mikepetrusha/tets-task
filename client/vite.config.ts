import react from '@vitejs/plugin-react';
import path from 'path';
import { defineConfig } from 'vite';
import EnvironmentPlugin from 'vite-plugin-environment';

// https://vitejs.dev/config/
export default defineConfig({
  resolve: {
    alias: {
      '@': path.resolve(__dirname, './src/')
    }
  },
  plugins: [react(), EnvironmentPlugin('all', { prefix: 'REACT_APP_' })],
  server: {
    port: 3000,
    open: true
  },
  define: {
    global: 'window'
  }
});
