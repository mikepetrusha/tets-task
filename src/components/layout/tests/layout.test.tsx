import { act } from '@testing-library/react';
import { createRoot, Root } from 'react-dom/client';
import renderer from 'react-test-renderer';

import { App } from '@/App';

import { Layout } from '../layout';

it('renders correctly', () => {
  const tree = renderer.create(<Layout />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders App correctly', () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
});
