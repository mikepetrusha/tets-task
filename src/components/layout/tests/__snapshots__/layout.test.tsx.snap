// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
[
  <header
    className="sc-beySPh kSpHrS"
  >
    <div
      className="sc-guDLey fHbHWv"
    >
      <h1
        className="sc-dmyCSP eBOhir"
      >
        ADRESSBUCH
      </h1>
    </div>
  </header>,
  <main>
    <div
      className="sc-eTNRI eHkrbZ"
      data-testid="test-spinner"
    >
      <div
        className="sc-brSamD iVLbrY"
      />
    </div>
  </main>,
  <div />,
]
`;

exports[`renders correctly 1`] = `
[
  <header
    className="sc-beySPh cOeJqa"
  >
    <div
      className="sc-guDLey dQJRti"
    >
      <h1
        className="sc-dmyCSP hYeZXF"
      >
        ADRESSBUCH
      </h1>
    </div>
  </header>,
  <main />,
  <div />,
]
`;
